core/include/traccc/seeding/detail/spacepoint_formation.hpp

```diff
@@ -16,12 +16,14 @@ namespace traccc::details {
 
 /// Function helping with filling/setting up a spacepoint object
 ///
-/// @param[in]  det         The tracking geometry
-/// @param[in]  measurement The measurement to create the spacepoint out of
+/// @param[in]   det          The tracking geometry
+/// @param[in]   measurement  The input measurement
+/// @param[out]  spacepoints  The spacepoint vector to be filled
 ///
-template <typename detector_t>
-TRACCC_HOST_DEVICE inline spacepoint create_spacepoint(const detector_t& det,
-                                                       const measurement& meas);
+template <typename detector_t, template <typename...> class vector_t>
+TRACCC_HOST_DEVICE inline void fill_spacepoint(
+    const detector_t& det, const measurement& meas,
+    vector_t<spacepoint>& spacepoints);
```

Comment:
```suggestion
template <typename detector_t, typename sp_vector_t>
TRACCC_HOST_DEVICE inline void fill_spacepoint(
    const detector_t& det, const measurement& meas,
    sp_vector_t& spacepoints);
```

Reply:
Why does it need to be templated code at all? 🤔 With the current EDM I would write this function as:

```suggestion
template <typename detector_t>
TRACCC_HOST_DEVICE inline bool is_valid_spacepoint(
    const detector_t& det, const measurement& meas);
template <typename detector_t>
TRACCC_HOST_DEVICE inline void fill_spacepoint(
    const detector_t& det, const measurement& meas,
    spacepoint& spacepoint);
```

And then leave it up to the caller to push new default spacepoints into their containers. Note that this formalism will work with eventual SoA containers as well.

I personally am not a big fan of either of the templating methods, as the implementation will still be AoS or SoA specific. So we don't win much by being generic in the interface.

Reply:
I just added a function of `is_valid_measurement` and leave `create_spacepoint()` as it is for now.